logic/resolvable/ConcludableTest.java

```diff
@@ -310,4 +312,378 @@ public void test_conjunction_anonymous_variable_constraints_are_not_conflated()
+                new Pair("{ $x > 1; }", "{$y -1.0;}"),
+                new Pair("{ $x > 1.0; }", "{$y -1;}"),
+                new Pair("{ $x > 1.0; }", "{$y -1.0;}"),
+
+                new Pair("{ $x > 1; }", "{$y 1;}"),
+                new Pair("{ $x > 1; }", "{$y 1.0;}"),
+                new Pair("{ $x > 1.0; }", "{$y 1;}"),
+                new Pair("{ $x > 1.0; }", "{$y 1.0;}"),
+
+                new Pair("{ $x < 1; }", "{$y 1;}"),
+                new Pair("{ $x < 1; }", "{$y 1.0;}"),
+                new Pair("{ $x < 1.0; }", "{$y 1;}"),
+                new Pair("{ $x < 1.0; }", "{$y 1.0;}"),
+
+                new Pair("{ $x < 1; }", "{$y 2;}"),
+                new Pair("{ $x < 1; }", "{$y 2.0;}"),
+                new Pair("{ $x < 1.0; }", "{$y 2;}"),
+                new Pair("{ $x < 1.0; }", "{$y 2.0;}"),
+
+                // String comparisons
+                new Pair("{ $x \"one\"; }", "{$y \"two\";}"),
+                new Pair("{ $x = \"one\"; }", "{$y \"two\";}"),
+                new Pair("{ $x >= \"two\"; }", "{$y \"one\";}"),
+                new Pair("{ $x <= \"one\"; }", "{$y \"two\";}"),
+                new Pair("{ $x != \"one\"; }", "{$y \"one\";}"),
+                new Pair("{ $x > \"one\"; }", "{$y \"one\";}"),
+                new Pair("{ $x > \"two\"; }", "{$y \"one\";}"),
+                new Pair("{ $x < \"two\"; }", "{$y \"two\";}"),
+                new Pair("{ $x < \"one\"; }", "{$y \"two\";}"),
+
+                // String comparisons, case-sensitivity
+                new Pair("{ $x \"OnE\"; }", "{$y \"oNe\";}"),
+                new Pair("{ $x = \"OnE\"; }", "{$y \"oNe\";}"),
+                new Pair("{ $x > \"two\"; }", "{$y \"ONE\";}"),
+                new Pair("{ $x < \"ONE\"; }", "{$y \"two\";}"),
+
+                // DateTime
+                new Pair("{ $x 2022-01-01; }", "{$y  2022-02-02;}"),
+                new Pair("{ $x = 2022-01-01; }", "{$y  2022-02-02;}"),
+                new Pair("{ $x != 2022-01-01; }", "{$y  2022-01-01;}"),
+
+                new Pair("{ $x >= 2022-02-02; }", "{$y 2022-01-01;}"),
+                new Pair("{ $x <= 2022-01-01; }", "{$y 2022-02-02;}"),
+
+                new Pair("{ $x > 2022-01-01; }", "{$y 2022-01-01;}"),
+                new Pair("{ $x > 2022-02-02; }", "{$y 2022-01-01;}"),
+                new Pair("{ $x < 2022-01-01; }", "{$y 2022-01-01;}"),
+                new Pair("{ $x < 2022-01-01; }", "{$y 2022-02-02;}"),
+        };
+
+        for (Pair<String, String> pair : constraintValuePairs) {
+            Concludable constraint = iterate(Concludable.create(parseConjunction(pair.first()))).next();
+            Concludable value = iterate(Concludable.create(parseConjunction(pair.second()))).next();
+
+            assertTrue(constraint.isAttribute());
+            assertTrue(value.isAttribute());
+            assertFalse(Unifier.Builder.unificationSatisfiable(constraint.asAttribute().attribute(), value.asAttribute().attribute()));
+        }
```

Comment:
Let's move this into `UnifierTest` as a unit test, and then not create Concludables - get the variables used for satisfiability tests from the conjunction directly